src/components/app-map/app-map.vue

```diff
@@ -25,16 +28,117 @@
     data() {
       return {
         mapCenter: [ 4.604, 52.467 ],
+        mapLoaded: false,
         mapToken: process.env.VUE_APP_MAPBOX_TOKEN,
-        mapZoom: 14.5,
+        mapZoom: 12.5,
+        shouldZoomIn: false,
+        popup: null,
       };
     },
     created() {
       this.mapbox = Mapbox;
+      this.popup = new Mapbox.Popup({
+        closeButton: false,
+        closeOnClick: false,
+        offset: 10,
+      });
+    },
+    computed: {
+      ...mapGetters('locations', [ 'locations' ]),
     },
     methods: {
+      ...mapActions('locations', [ 'setActiveLocation' ]),
+      addListeners() {
+        this.$root.map.on('mouseenter', 'markers', this.onMouseEnter);
+        this.$root.map.on('mouseleave', 'markers', this.onMouseLeave);
+      },
       onMapCreated({ map }) {
         this.$root.map = map;
+        this.mapLoaded = true;
+        this.shouldZoomIn = true;
+      },
+      onClickMarker(id) {
+        this.setActiveLocation({ id });
+      },
+      onMouseEnter(event) {
+        const { features, lngLat } = event;
+        const coordinates = features[0].geometry.coordinates.slice();
+        const { loc_id, meanhead } = features[0].properties;
+        const content = `<dl>
+          <dt class="text-subtitle-2">Locatie id:</dt> <dd class="text-body-2">${ loc_id }</dd>
+          <dt class="text-subtitle-2">Gem. gwstand:</dt> <dd class="text-body-2">${ meanhead }</dd>
+        </dl>`;
+
+        // Change the cursor style as a UI indicator.
+        this.$root.map.getCanvas().style.cursor = 'pointer';
+
+        // Ensure that if the map is zoomed out such that multiple
+        // copies of the feature are visible, the popup appears
+        // over the copy being pointed to.
+        while (Math.abs(lngLat.lng - coordinates[0]) > 180) {
+          coordinates[0] += lngLat.lng > coordinates[0] ? 360 : -360;
+        }
+
+        // Populate the popup and set its coordinates
+        // based on the feature found.
+        this.popup
+          .setLngLat(coordinates)
+          .setHTML(content)
+          .addTo(this.$root.map);
+      },
+      onMouseLeave() {
+        this.$root.map.getCanvas().style.cursor = '';
+        this.popup.remove();
+      },
+      populateMap() {
+        this.$root.map.addSource('markers', {
+          type: 'geojson',
+          data: featureCollection(
+            this.locations.map((location) => ({
+              geometry: location.geometry,
+              properties: location.properties,
+              type: 'Feature',
+            }))
+          ),
+        });
+
+        // Add a layer showing the places.
+        this.$root.map.addLayer({
+          id: 'markers',
+          type: 'circle',
+          source: 'markers',
+          paint: {
+            'circle-color': '#fff',
+            'circle-radius': 5,
+            'circle-stroke-width': 5,
+            'circle-stroke-color': '#ff0000',
+          },
+        });
+      },
```

Comment:
Please extract the markers to a separate component

Reply:
example: https://github.com/openearth/vue2mapbox-gl/blob/master/src/components/v-mapbox-layer.js